js-src/bindings.ts

```diff
@@ -217,11 +217,17 @@ export interface SignClaimBytesProps {
   signer: Signer;
 }
 
-export interface SignOutput {
-  signedAsset: Asset;
+export interface SignOutputData<AssetType extends Asset = Asset> {
+  signedAsset: AssetType;
   signedManifest?: Buffer;
 }
 
+export type SignOutput<AssetType> = AssetType extends BufferAsset
+  ? SignOutputData<BufferAsset>
+  : AssetType extends FileAsset
+  ? SignOutputData<FileAsset>
+  : never;
+
```

Comment:
The changes in this file take advantage of conditional types to make the output type of the `sign` function explicit, rather than part of the `AssetType` union - if buffer data is provided, the output type is a `BufferAsset,` etc. 

Without this the consumer needs to perform checks (e.g. `'buffer' in asset`) to narrow the type.

Reply:
Awesome!